src/server/pps/server/worker_rc.go

```diff
@@ -95,8 +95,18 @@ func (a *apiServer) workerPodSpec(options *workerOptions) (v1.PodSpec, error) {
 		userVolumeMounts = append(userVolumeMounts, storageMount)
 	}
 	secretVolume, secretMount := assets.GetBackendSecretVolumeAndMount(a.storageBackend)
-	options.volumes = append(options.volumes, secretVolume)
-	sidecarVolumeMounts = append(sidecarVolumeMounts, secretMount)
+	// `pach-dir-volume` is needed for openshift, see:
+	// https://github.com/pachyderm/pachyderm/issues/3404
+	options.volumes = append(options.volumes, secretVolume, v1.Volume{
+		Name: "pach-dir-volume",
+		VolumeSource: v1.VolumeSource{
+			EmptyDir: &v1.EmptyDirVolumeSource{},
+		},
+	})
+	sidecarVolumeMounts = append(sidecarVolumeMounts, secretMount, v1.VolumeMount{
+		Name:      "pach-dir-volume",
+		MountPath: a.storageRoot,
+	})
```

Comment:
Playing around with this on my machine I get the following when I try to run `TestSimplePipeline`:

```
error running the master process: ReplicationController "pipeline-testsimplepipelinec5ed94a91679-v1" is invalid: spec.template.spec.containers[1].volumeMounts[2].mountPath: Invalid value: "/pach": must be unique; retrying in 3.6971187s"
```

Which makes sense because in local deployments we put a hostpath mount there, so we get an overlap. The place where we do this is a few lines above (starts on 81), I think if you put this in an `else` block for that conditional it would pass CI. There's still the open question as to where that `MkdirAll` is coming from that's creating the error. I did some more checking and can't figure it out. It might be a good idea to run a binary with a `runtime.PrintBacktrace()` call next to the error handling for that `MkdirAll` just so we can see the execution path that's leading to it.